api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
In the test :- 
before the change:- all the new tests pass
after the test:- the last 2 new tests fail since new obs have been created to replace the old one